python/ray/tune/trial.py

```diff
@@ -326,6 +329,20 @@ def update_last_result(self, result, terminate=False):
         self.last_result = result
         self.last_update_time = time.time()
         self.result_logger.on_result(self.last_result)
+        for metric, value in result.items():
+            if (value is not None) and (type(value) is not dict):
+                if metric not in self.metric_analysis.keys():
```

Comment:
nit: remove parenthesis in L#333, also no need to call `.keys()` on the dict i think